File: python/225.py
import queue as Queue


class Stack(object):
    def __init__(self):
        """
        initialize your data structure here.
        """
        self.q1 = Queue.Queue()
        self.q2 = Queue.Queue()

    def push(self, x):
        """
        :type x: int
        :rtype: nothing
        """
        self.q1.put(x)

    def pop(self):
        """
        :rtype: nothing
        """
        for i in range(self.q1.qsize()-1):
            self.q2.put(self.q1.get())
        self.q1, self.q2 = self.q2, self.q1
        self.q2.get()

    def top(self):
        """
        :rtype: int
        """
        last = None
        for i in range(self.q1.qsize()):
            last = self.q1.get()
            self.q2.put(last)
        self.q1, self.q2 = self.q2, self.q1
        return last

    def empty(self):
        """
        :rtype: bool
        """
        return self.q1.empty()

File: cpp/225.cpp
#include <queue>
#include <iostream>
#include <algorithm>
using namespace std;

class Stack {
  queue<int> q1, q2;

public:
  // Push element x onto stack.
  void push(int x) { q1.push(x); }

  // Removes the element on top of the stack.
  void pop() {
    int n = q1.size();
    for (size_t i = 0; i < n - 1; i++) {
      q2.push(q1.front());
      q1.pop();
    }
    q1.pop();
    swap(q1, q2);
  }

  // Get the top element.
  int top() { return q1.back(); }

  // Return whether the stack is empty.
  bool empty() { return q1.empty(); }
};

int main() {
  auto s = Stack();
  s.push(1);
  s.push(2);
  s.push(3);
  cout << s.top() << endl;
  s.pop();
  cout << s.top() << endl;
}
